test/Microsoft.ML.Tests/TrainerEstimators/TreeEnsembleFeaturizerTest.cs

```diff
@@ -108,5 +115,700 @@ public void TreeEnsembleFeaturizerOutputSchemaTest()
+                var leafId = model.Model.SubModel.GetLeaf(treeIndex, new VBuffer<float>(10, features[dataPointIndex]), ref path);
+                var leafValue = model.Model.SubModel.GetLeafValue(0, leafId);
+                Assert.Equal(leafValues[dataPointIndex][treeIndex], leafValue);
+                Assert.Equal(1.0, leafIds[dataPointIndex][leafId]);
+                foreach (var nodeId in path)
+                    Assert.Equal(1.0, paths[dataPointIndex][nodeId]);
+            }
+        }
+
+        /// <summary>
+        /// This test contains several steps.
+        ///   1. It first trains a <see cref="FastTreeBinaryModelParameters"/> using <see cref="FastTreeBinaryTrainer"/>.
+        ///   2. Then, it creates the a <see cref="PretrainedTreeFeaturizationEstimator"/> from the trained <see cref="FastTreeBinaryModelParameters"/>.
+        ///   3. The feature produced in step 2 would be fed into <see cref="SdcaLogisticRegression"/> to enhance the training accuracy of that linear model.
+        ///   4. We train another <see cref="SdcaLogisticRegression"/> without features from trees and finally compare their scores.
+        /// </summary>
+        [Fact]
+        public void TreeEnsembleFeaturizingPipeline()
+        {
+            // Create data set
+            int dataPointCount = 200;
+            var data = SamplesUtils.DatasetUtils.GenerateBinaryLabelFloatFeatureVectorFloatWeightSamples(dataPointCount).ToList();
+            var dataView = ML.Data.LoadFromEnumerable(data);
+            dataView = ML.Data.Cache(dataView);
+
+            // Define a tree model whose trees will be extracted to construct a tree featurizer.
+            var trainer = ML.BinaryClassification.Trainers.FastTree(
+                new FastTreeBinaryTrainer.Options
+                {
+                    NumberOfThreads = 1,
+                    NumberOfTrees = 10,
+                    NumberOfLeaves = 4,
+                    MinimumExampleCountPerLeaf = 10
+                });
+
+            // Train the defined tree model. This trained model will be used to construct TreeEnsembleFeaturizationEstimator.
+            var treeModel = trainer.Fit(dataView);
+            var predicted = treeModel.Transform(dataView);
+
+            // Combine the output of TreeEnsembleFeaturizationTransformer and the original features as the final training features.
+            // Then train a linear model.
+            var options = new PretrainedTreeFeaturizationEstimator.Options()
+            {
+                InputColumnName = "Features",
+                TreesColumnName = "Trees",
+                LeavesColumnName = "Leaves",
+                PathsColumnName = "Paths",
+                ModelParameters = treeModel.Model.SubModel
+            };
+            var pipeline = ML.Transforms.FeaturizeByPretrainTreeEnsemble(options).
+                Append(ML.Transforms.Concatenate("CombinedFeatures", "Features", "Trees", "Leaves", "Paths")).
```

Comment:
Isn't the normal style to have a period at the start?
```suggestion
                .Append(ML.Transforms.Concatenate("CombinedFeatures", "Features", "Trees", "Leaves", "Paths"))
```

Reply:
You're right. Let me fix them. Thanks.